aspnetcore/grpc/diagnostics.md

```diff
@@ -99,6 +107,88 @@ dbug: Grpc.Net.Client.Internal.GrpcCall[4]
       Finished gRPC call.
 ```
 
+## Tracing
+
+gRPC services and the gRPC client provide information about gRPC calls using [DiagnosticSource](https://docs.microsoft.com/dotnet/api/system.diagnostics.diagnosticsource) and [Activity](https://docs.microsoft.com/dotnet/api/system.diagnostics.activity). .NET gRPC uses an activity to represent a gRPC call. Tracing events are written to the diagnostic source at the start and stop of the gRPC call activity.
+
```

Comment:
What happens with streaming? I believe there is one looong request and individual events/messages are not instrumented. I guess it's useful to mention it and suggest to add custom tracing for individual events/messages

Reply:
Also worth mentioning that context is propagated over HTTP headers following https://www.w3.org/TR/trace-context-1/ when tracing is enabled.

Reply:
Yes there is one long activity for the streaming call. New activities for reading or writing individual messages didn't make sense to me because they happen at an instant in time. There is no begin and end.

gRPC writes to ILogger when messages are read/written during streaming. I can call that out. Do you have a suggestion for additional instrumentation?

Reply:
I don't know enough about gRPC, but my guess that with streaming, you'd have one long request for something like app lifetime (minutes, hours, days) and individual messages originate from the different contexts and are not related in any way. Is my assumption correct?

If it is, and if this scenario is popular, we may need to design better support for it (and similar scenarios). I recall signalR users have the same concern and HttpIn Activities makes no sense to them. It would also mean context propagation for each message (probably an overkill).

If streaming is usually short and individual messages are coming from the same tracing context on the upstream, I believe what we have today is enough.